packages/blockchain-wallet-v4-frontend/src/layouts/Dex/DexHeader/DexHeader.tsx

```diff
@@ -1,103 +1,207 @@
+      dropdownSecondSectionItems={[
+        {
+          key: 'general',
+          label: formatMessage({
+            defaultMessage: 'General',
+            id: 'navbar.dropdown.general'
+          })
+        },
+        {
+          key: 'security',
+          label: formatMessage({
+            defaultMessage: 'Security',
+            id: 'navbar.dropdown.security'
+          })
+        },
+        {
+          key: 'trading-limits',
+          label: formatMessage({
+            defaultMessage: 'Trading Limits',
+            id: 'navbar.dropdown.tradingLimits'
+          })
+        },
+        {
+          key: 'preferences',
+          label: formatMessage({
+            defaultMessage: 'Preferences',
+            id: 'navbar.dropdown.preferences'
+          })
+        },
+        {
+          key: 'wallets-addresses',
+          label: formatMessage({
+            defaultMessage: 'Wallets & Addresses',
+            id: 'navbar.dropdown.walletsAndAddresses'
+          })
+        },
+        {
+          key: 'refer-friend',
+          label: formatMessage({
+            defaultMessage: 'Refer a friend',
+            id: 'navbar.dropdown.referFriend'
+          })
+        },
+        {
+          key: 'tax-center',
+          label: formatMessage({
+            defaultMessage: 'Tax Center',
+            id: 'navbar.dropdown.taxCenter'
+          })
+        }
+      ]}
+      dropdownSecondSectionSeparator={{
+        key: 'account',
+        label: formatMessage({
+          defaultMessage: 'Account',
+          id: 'navbar.dropdown.separator'
+        })
+      }}
+      iconActions={[
+        {
+          icon: () => <IconRefresh />,
+          label: formatMessage({
+            defaultMessage: 'Refresh',
+            id: 'navbar.icons.refresh'
+          }),
+          // TODO: Handle refresh
+          onClick: () => undefined
+        },
+        {
+          // TODO: Handle app QR code dropdown
+          icon: () => <IconPhone />,
+
+          label: formatMessage({
+            defaultMessage: 'Phone app',
+            id: 'navbar.icons.phoneApp'
+          }),
+          onClick: () => undefined
+        }
+      ]}
+      // TODO: Fetch correct information and handle a click
+      walletButton={{
+        id: '14qViLJfdGaP4EeHnDyJbEGQysnCpwk3gd',
+        imgAlt: 'ETH',
+        imgSrc:
+          'https://cloudfront-us-east-1.images.arcpublishing.com/coindesk/ZJZZK5B2ZNF25LYQHMUTBTOMLU.png',
+        onClick: () => undefined
+      }}
+      // TODO: Pass real user details
+      user={{
+        name: 'John Doe',
+        // TODO: Handle dropdown
+        onClick: () => undefined
+      }}
+    />
   )
 }
 
-const mapStateToProps = (state) => ({
-  ethBalanceR: selectors.balances.getCoinNonCustodialBalance('ETH')(state)
-})
-
-const connector = connect(mapStateToProps)
-type Props = ConnectedProps<typeof connector> & OwnProps
-
-export default connector(DexHeader)
+export const DexHeader = withRouter<Props>(Header)
```

Comment:
After our conversation I looked into this a little more. I think instead of wrapping the component in `withRouter` and using `history.push()` you can just dispatch a route change action like this: `dispatch(actions.router.push('/home'))` you can see a few examples in the code if you grep for that string.

Reply:
Thanks, that works! Fixed